shims/spark310/src/main/scala/org/apache/parquet/hadoop/ParquetOutputFileFormat.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.parquet.hadoop
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.mapreduce.RecordWriter
+import org.apache.parquet.column.ParquetProperties
+import org.apache.parquet.hadoop.ParquetFileWriter.Mode
+import org.apache.parquet.hadoop.api.WriteSupport
+import org.apache.parquet.hadoop.metadata.CompressionCodecName
+import org.apache.parquet.io.OutputFile
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.datasources.parquet.ParquetWriteSupport
+
+class ParquetOutputFileFormat {
+
+  private var memoryManager: MemoryManager = null
+  val writeSupport = new ParquetWriteSupport().asInstanceOf[WriteSupport[InternalRow]]
+  val DEFAULT_MEMORY_POOL_RATIO: Float = 0.95f
+  val DEFAULT_MIN_MEMORY_ALLOCATION: Long = 1 * 1024 * 1024 // 1MB
+
+  def getRecordWriter(output: OutputFile, conf: Configuration): RecordWriter[Void, InternalRow] = {
+    import ParquetOutputFormat._
+
+    val blockSize = getLongBlockSize(conf)
+    val maxPaddingSize =
+      conf.getInt(MAX_PADDING_BYTES, ParquetWriter.MAX_PADDING_SIZE_DEFAULT)
+    val validating = getValidation(conf)
+
+    val init = writeSupport.init(conf)
+    val w = new ParquetFileWriter(output, init.getSchema,
+      Mode.CREATE, blockSize, maxPaddingSize)
+    w.start()
+
+    val maxLoad = conf.getFloat(MEMORY_POOL_RATIO, DEFAULT_MEMORY_POOL_RATIO)
+    val minAllocation = conf.getLong(MIN_MEMORY_ALLOCATION, DEFAULT_MIN_MEMORY_ALLOCATION)
+
+    classOf[ParquetOutputFormat[InternalRow]].synchronized {
```

Comment:
Synchronizing on another class to initialize a member of this class that doesn't seem to need the other class to initialize the value is quite odd, and I'm not seeing the purpose of it.

Why isn't `memoryManager` simply a local variable of this method?  I don't see it used outside of this method, and the code clobbers the old value each time this method is called.  Is it not supposed to initialize this more than once?  If so, I'd recommend using a `lazy val` instead.

Reply:
Thanks, I have replaced it with lazy val